@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -1358,6 +1356,9 @@ export class TileGeometryCreator {
      * @param renderOrder - Render order of the tile
      */
     addGroundPlane(tile: Tile, renderOrder: number) {
+        if (!tile.dataSource.isFullyCovering || tile.projection.type !== ProjectionType.Planar) {
```

Comment:
Actually looking at this a second time, I think, moving this check inside the method is error prone, it covers up information, that will lead to misusage and a lot of debugging :) .. it should be self contained and the code around decide to call it or not

Reply:
I guess it is a compromise, by moving the logic outside of this function, you may also introduce misusage, because the function may be called with different checks... and someone who wants to read the code would need to understand why there are different checks. For example, if this was fixed to work with sphere projection, then the person would need to find all calls to this function and update the external checks to remove the check for the Planar type (which they may miss) Easier if the ground plane logic is in one location and not spread in different places IMO.

Reply:
@FraukeF , I changed the logic, so this change is no longer necessary, I hope to push the new changes soon.